LiteCore/tests/LazyVectorAPITest.cc

```diff
@@ -0,0 +1,354 @@
+        auto& store = asInternal(db)->dataFile()->defaultKeyStore();
+        try {
+            return make_retained<LazyIndex>(store, name);
+        } catch ( [[maybe_unused]] std::exception& e ) { return nullptr; }
+    }
+
+    void checkQueryReturnsWords(C4Query* query, const std::vector<string>& expectedWords) const {
+        auto e = REQUIRED(c4query_run(query, _encodedTarget, ERROR_INFO()));
+        REQUIRE(c4queryenum_getRowCount(e, ERROR_INFO()) == expectedWords.size());
+        for ( size_t i = 0; i < expectedWords.size(); ++i ) {
+            INFO("i=" << i);
+            REQUIRE(c4queryenum_next(e, ERROR_INFO()));
+            FLArrayIterator columns  = e->columns;
+            slice           word     = Value(FLArrayIterator_GetValueAt(&columns, 0)).asString();
+            float           distance = Value(FLArrayIterator_GetValueAt(&columns, 1)).asFloat();
+            Log("%.*s: %.3f", FMTSLICE(word), distance);
+            CHECK(word == slice(expectedWords[i]));
+            // CHECK(fabs(distance - expectedDistances[i]) < 0.01);
+        }
+        CHECK(!c4queryenum_next(e, ERROR_INFO()));
+        Log("done");
+        c4queryenum_release(e);
+    }
+
+    void checkQueryReturnsVectors(C4Query* query, int64_t expectedRowCount,
+                                  const std::vector<float>& expectedVectors) const {
+        auto e = REQUIRED(c4query_run(query, _encodedTarget, ERROR_INFO()));
+        REQUIRE(c4queryenum_getRowCount(e, ERROR_INFO()) == expectedRowCount);
+        for ( size_t i = 0; i < expectedRowCount; ++i ) {
+            INFO("i=" << i);
+            REQUIRE(c4queryenum_next(e, ERROR_INFO()));
+            FLArrayIterator columns     = e->columns;
+            auto            vectorArray = Value(FLArrayIterator_GetValueAt(&columns, 0)).asArray();
+            for ( size_t j = 0; j < expectedVectors.size(); j++ ) {
+                float vector = vectorArray[j].asFloat();
+                CHECK(vector == expectedVectors[j]);
+            }
+        }
+        CHECK(!c4queryenum_next(e, ERROR_INFO()));
+        Log("done");
+        c4queryenum_release(e);
+    }
+
+    bool createIndex(slice name, slice jsonSpec, C4IndexType type, C4IndexOptions options, C4Error* err) const {
+        return c4coll_createIndex(_wordsColl, name, jsonSpec, kC4JSONQuery, type, &options, err);
+    }
+
+    bool createVectorIndex(bool lazy, slice expression = R"(['.word'])"_sl, slice name = "words_index"_sl,
+                           C4Error* err = ERROR_INFO()) const {
+        auto options = vectorOptions(300, 8);
+        options.lazy = lazy;
+        return createIndex(name, json5(expression), kC4VectorIndex, indexOptions(options), err);
+    }
+
+    C4Index* getIndex(slice name = "words_index"_sl, C4Error* err = ERROR_INFO()) const {
+        return c4coll_getIndex(_wordsColl, name, err);
+    }
+
+    void createVectorDocs(unsigned numberOfDocs) const {
```

Comment:
It seems like there are many functions in this class are not being used? Are these for another PR?

Reply:
Yeah. I copied the class from my branch for lazy index testing.